src/ensembl/src/content/app/browser/track-panel/track-panel-list/TrackPanelListItem.tsx

```diff
@@ -139,42 +177,29 @@ export const TrackPanelListItem = (props: TrackPanelListItemProps) => {
     const newStatus =
       trackStatus === Status.ACTIVE ? Status.INACTIVE : Status.ACTIVE;
 
-    updateGenomeBrowser(newStatus);
-
-    if (!activeGenomeId || !activeEnsObjectId) {
-      return;
-    }
-    // FIXME: Temporary hack until we have a set of proper track names
+    // FIXME: Temporary hack to toggle the focus track either using the eye icon next to the gene or the one next transcript.
     if (track.track_id.startsWith('track:gene')) {
-      props.updateTrackStatesAndSave({
-        [activeGenomeId]: {
-          objectTracks: {
-            [activeEnsObjectId]: {
-              [categoryName]: {
-                [track.track_id]: newStatus
-              }
-            }
-          }
-        }
-      });
-    } else {
-      props.updateTrackStatesAndSave({
-        [activeGenomeId]: {
-          commonTracks: {
-            [categoryName]: {
-              [track.track_id]: newStatus
-            }
-          }
-        }
-      });
+      updateTrackStates('track:gene-feat', newStatus);
+      // Make sure we have a track with id 'track:gene-feat-1' before setting a status.
+      if (
+        track.track_id === 'track:gene-feat-1' ||
+        (track.child_tracks && track.child_tracks.length)
+      ) {
+        updateTrackStates('track:gene-feat-1', newStatus);
+      }
+      updateGenomeBrowser('track:gene-feat', newStatus);
+      return;
```

Comment:
Think it'll be easier to understand what is going on here if this `return` was replaced by an `else` statement